src/general/VersionUtils.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Helpers for dealing with Thrive's version number
+/// </summary>
+public static class VersionUtils
+{
+    public static readonly string[] Suffixes =
+    {
+        "pre-alpha",
+        "alpha",
+        "beta",
+        "rc1",
+        "rc2",
+        "rc3",
+        "rc4",
+        "rc5",
+        "rc6",
+        "rc7",
+        "rc8",
+        "rc9",
+    };
+
+    /// <summary>
+    ///   Compare the given version numbers.
+    /// </summary>
+    /// <returns>
+    ///   0 if the versions are the same,
+    ///   a negative integer if a is a smaller (older) version than b and
+    ///   a positive integer if a is a bigger (newer) version than b
+    /// </returns>
+    /// <param name="a">The first version to compare.</param>
+    /// <param name="b">The second version to compare.</param>
+    public static int Compare(string a, string b)
+    {
+        if (a == b)
+        {
+            return 0;
+        }
+
+        char[] separator = { '-' };
+        var aSplit = a.Split(separator, 2);
+        var bSplit = b.Split(separator, 2);
+
+        // Compare the numeric versions
+        int versionDiff = Version.Parse(aSplit[0]).CompareTo(Version.Parse(bSplit[0]));
```

Comment:
I think there should be a catch here, which logs an exception for parsing failure and returns 0 (to indicate they can't be ordered).
Just in case there exists some really funky version numbers that need comparing. It wouldn't be very nice to crash the game when opening the save list (which can happen when trying to save the game).